### PR TITLE
Android openssl atomic bugfix with ndk update

### LINKED ISSUE #4696: [Android][vcpkg] Remove vcpkg pinning
Currently there is a major issue with Android and OpenSSL 3.0.2:
- Android vcpkg build is failed because of clang issue (https://github.com/openssl/openssl/issues/18059). It would be possible to workaround this if https://github.com/openssl/openssl/pull/18056 is merged but still using custom port only.

When https://github.com/microsoft/vcpkg/pull/24004 and https://github.com/openssl/openssl/pull/18056 are merged, new port should be created to properly fix that in vcpkg upstream

### PR DESCRIPTION
**Pull request by talregev**
Android openssl atomic bugfix with ndk update.
Fixes https://github.com/microsoft/vcpkg/issues/24021 
Fixes #4696

### CODE REVIEW

**Comment by talregev** [comment]
@AenBleidd This is fix the openssl problem atomic with update of clang (update ndk). I think this fix is better, and we should cancel your fix to vcpkg.

**Comment by talregev** [comment]
@AenBleidd This fix interduce a new problem. That vcpkg compile arm with neon, even if the neon set to False.
I want to investigate this problem. It ok if you put your vcpkg PR on draft?

File: android/ci_build_libs_cmake.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ for TRIPLET in $TRIPLETS_LIST ; do
         export ANDROID_NDK_HOME=$NDK_ROOT
     fi
     BUILD_TRIPLET=build-$TRIPLET
-    cmake lib -B $BUILD_TRIPLET -DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake -DVCPKG_OVERLAY_PORTS=$VCPKG_PORTS/ports -DVCPKG_OVERLAY_TRIPLETS=$VCPKG_PORTS/triplets/ci -DVCPKG_TARGET_TRIPLET=$TRIPLET -DVCPKG_MANIFEST_DIR=lib/vcpkg_config_android -DVCPKG_INSTALL_OPTIONS=--clean-after-build
+    cmake lib -B $BUILD_TRIPLET -DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake -DVCPKG_OVERLAY_PORTS=$VCPKG_PORTS/ports -DVCPKG_OVERLAY_TRIPLETS=$VCPKG_PORTS/triplets/ci -DVCPKG_TARGET_TRIPLET=$TRIPLET -DVCPKG_INSTALL_OPTIONS=--clean-after-build
```
Comment (AenBleidd) [inline android/ci_build_libs_cmake.sh:38]: Why did you remove this manifest file?

Comment (talregev) [reply android/ci_build_libs_cmake.sh:38]: Because we now can use the same manifest cmake file for all os. (linux, windows and android).

**Review by AenBleidd — APPROVED** [review]

**Pull request merged**